### PR TITLE
Shape prop for radio and checkbox buttons

### PR DESCRIPTION
**Pull request by PickertJoe**
#### What does this PR do?
This PR is a first step towards reducing some of the code duplication across our different radio button shapes. Additionally, it adds the same flexibility to our existing `CheckboxButton` component to prevent this kind of duplication from recurring if we need checkboxes of different shapes.

While `CheckboxButton` is used in other components, the new `shape` prop is optional and the checkbox will default to its previous size/shape if it isn't provided.

When working on the mini quiz, I noticed that these components had essentially been recreated with minimal changes in several instances, and it was difficult to know which one I should use. In making these buttons more flexible, I'm hoping that we can treat these components as sources of truth across our projects when we need to use them in forms. 

I think `size` would be another helpful prop to add to these buttons, but am curious to hear your thoughts!

Another note: my idea is to change the `RectangularRadioButton` component to just be our default `RadioButton` component with a `shape` prop once I've finished updating everything that currently depends on the `RoundRadioButton` and `SquarRadioButton` components

### CODE REVIEW

File: src/components/inputs/CheckboxButton/CheckboxButton.js [hunk 1]
```diff
@@ -1,5 +1,60 @@
-import styled from 'styled-components'
-import BaseCheckboxButton from './CheckboxButton.base'
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled, { css } from 'styled-components'
+import ErrorMessage from 'SRC/components/inputs/ErrorMessage'
+
+const BaseCheckboxButton = ({className, input, value, error, children, ...props}) => {
```
Comment (manmartinez) [inline src/components/inputs/CheckboxButton/CheckboxButton.js:6]: Should we remove the `input` prop here?

File: src/components/inputs/CheckboxButton/CheckboxButton.js [hunk 2]
```diff
@@ -1,5 +1,60 @@
...
+      </label>
+      {error &&
+        <ErrorMessage>{error}</ErrorMessage>
+      }
+    </div>
+  )
+}
+
+const handleShapeStyles = shape => {
+  switch (shape) {
```
Comment (manmartinez) [inline src/components/inputs/CheckboxButton/CheckboxButton.js:21]: Perhaps we could simplify this by using a hash? Something like:

```js
const squareStyle = css`
  height: 12rem;
  width: 12rem;
`

const rectangleStyle = css`
  height: 6.4rem;
  width: 100%;
`

const shapeVariants = {
  square: squareStyle,
  rectangle: rectangleStyle,
  ...
}

function handleShapeStyles(shape) {
  return shapeVariants[shape]
}

File: src/components/inputs/CheckboxButton/CheckboxButton.js [hunk 3]
```diff
@@ -1,5 +1,60 @@
...
+
+function borderColor (props) {
+  if (props.error) {
+    return props.theme.colors.flameOrange
+  }
+  return props.theme.colors.gray4
+}
+
+const checked = css`
+  color: ${props => props.theme.colors.white};
```
Comment (manmartinez) [inline src/components/inputs/CheckboxButton/CheckboxButton.js:54]: Why do we need to have this CSS separate from the rest? It looks like we're not adding this conditionally, would it make sense to just inline it in line 89?

File: src/components/inputs/radioButtons/rectangularRadioButton.js [hunk 4]
```diff
@@ -24,6 +24,34 @@ function borderColor(props) {
   return props.theme.colors.gray4
 }
 
+const handleShapeStyles = shape => {
```
Comment (manmartinez) [inline src/components/inputs/radioButtons/rectangularRadioButton.js:27]: Why does the `RectangularRadioButton` need shapes too? Wouldn't we just use the `RadioButton` component with `shape='rectangular'`?

Comment (manmartinez) [reply src/components/inputs/radioButtons/rectangularRadioButton.js:27]: Duh I just realized one's a radio button and another one a checkbox, sorry about the confusion! 😅

File: src/components/inputs/CheckboxButton/CheckboxButton.js (hunk 2 above)
Comment (PickertJoe) [reply src/components/inputs/CheckboxButton/CheckboxButton.js:21]: I like that idea! Switch/case's always feel awkward to me

**Pull request merged**